### ISSUE TITLE
Why is `k2mask` set to use only half the number of available cores?

### ISSUE THREAD

**Issue by hermidalc** [DerrickWood/kraken2#858]
https://github.com/DerrickWood/kraken2/blob/4cbdc5fac92d0a19d76ce68d6633d7ff794a1587/scripts/k2#L485-L487

**Comment by ChillarAnand** [comment]
For `kraken2` I observed that performance is best at 1/2 to 3/4 of available cores. After that performance decreases.

Not sure about if it is the same case with `k2mask`. Need to run verify it.

https://avilpage.com/2024/07/mastering-kraken2-performance-optimisation.html

**Comment by hermidalc** [comment]
> For `kraken2` I observed that performance is best at 1/2 to 3/4 of available cores. After that performance decreases.
> 
> Not sure about if it is the same case with `k2mask`. Need to run verify it.
> 
> https://avilpage.com/2024/07/mastering-kraken2-performance-optimisation.html

The downside to it being hardcoded like that is when you run Kraken2 in a cluster environment you don't want the script looking at all the cores available on a node because you typically ask for a certain number of cores to use for the job and the job could be assigned to a node with a lot more cores (that other jobs are using).

**Comment by ChillarAnand** [comment]
I agree that it shouldn't be hardcoded. I am trying to understand why it could be hardcoded to `multiprocessing.cpu_count() // 2` in the initial stage.

**Comment by ch4rr0** [comment]
I agree that this change should be configurable via the command line. I will address this in my next commit to k2.

**Comment by ChillarAnand** [comment]
Raised PR for that https://github.com/DerrickWood/kraken2/pull/866
